server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -130,6 +135,31 @@ public ClusterState execute(ClusterState currentState) {
                 }
 
                 if (IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.exists(openSettings)) {
+
+                    final boolean updatingReplicasHasEffect;
+                    if (IndexMetadata.INDEX_AUTO_EXPAND_REPLICAS_SETTING.exists(openSettings)) {
```

Comment:
NB this doesn't account for updates that use wildcards, but nor does the rest of this logic.